### ISSUE TITLE
Sequential number generator wanted

### ISSUE THREAD

**Issue by dothebart** [arangodb/arangodb#2535]
Other databases offer sources for growing numbers (I.e. oracle sequences, mysql auto increment columns)

We should add a similar feature.

**Comment by hwde** [comment]
I think you could do this already with keyOptions, but not from GUI: create a collection with arangosh:

`db._create("mycollection", { "keyOptions":{"allowUserKeys":true,"type": "autoincrement"}});`

**Comment by dothebart** [comment]
@hwde Good catch! except for... `ArangoError 1470: non-traditional key generators are not supported for sharded collections` which is [missing in the Documentation](https://docs.arangodb.com/latest/Manual/DataModeling/Collections/DatabaseMethods.html)- fixed with 6bd5df42ea2495b389f8e60cca5cdcf77d7990cc

**Comment by akki** [comment]
Is it possible not to restrict it to insertions/updations. For example, what [`enumerate`](https://stackoverflow.com/a/22171642/3061686) does in Python.

I don't know how much this job is in the scope of a ArangoDB and AQL, but putting it up just as a suggestion to know if it can be implemented.

**Comment by jsteemann** [comment]
The problem is that we need something that works the same way in a distributed environment, e.g. use a sequence from multiple shards on different servers. Then this is not 100% simple if it should still be fast. But it sounds doable.

**Comment by dothebart** [comment]
Related / duplicate: https://github.com/arangodb/arangodb/issues/6041

**Comment by wajda** [comment]
Doing research about _autoincrement_ support on cluster I encountered two statements which are kinda contradicted to me:
1. "Error: 1470 - the specified key generator is not supported for sharded collections"
2. (from the doc) "In a cluster the autoincrement key generator is not supported"

Can you help me understand which is more correct and why?

I understand that computing sequential key values might be difficult on sharded collections, but if a collection only consists of a single shard with synced replicas, then from the perspective of autoincrementing keys it should behave in the same way as if it were a single server, shouldn't it?

**Comment by gurpreetbhatia** [comment]
@jsteemann,

As per ArangoDB docs, “autoincrement is currently only supported for non-sharded collections”. Refer https://www.arangodb.com/docs/stable/data-modeling-collections-database-methods.html.  **Do collections created in oneShard setup support autoincrement _key?** I could not find any such claim under OneShard documentation.

**Comment by rossmills99** [comment]
Any update on this?  The documentation around autoincrement is not very helpful here.  We would also like to use autoincrement keys in ArangoDB Oasis using the OneShard deployment type.  It seems it is not supported, can you at least confirm that?

**Comment by SoundsSerious** [comment]
+1 for this. Somewhat puzzling this feature wouldn't be supported for 1-shard on arango oasis since there isn't a less sharded option.

**Comment by SoundsSerious** [comment]
An easy solution for a `oneShard` deployment was to switch `autoincrement` to `traditional`
